Remove ProfiledType

ilia-cher opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#42570 Remove ProfiledType**

Summary:
ProfiledType doesn't do anything and is not used atm, removing

Test Plan: CI

Differential Revision: [D22938664](https://our.internmc.facebook.com/intern/diff/D22938664)

Commit:
Remove ProfiledType

Summary:
ProfiledType doesn't do anything and is not used atm, removing

Test Plan: CI

[ghstack-poisoned]

ezyang reviewed (approved):
This makes my day! Thank you!!

bhosmer reviewed (approved):
🎉

Commit:
Update on "Remove ProfiledType"


Summary:
ProfiledType doesn't do anything and is not used atm, removing

Test Plan: CI

Differential Revision: [D22938664](https://our.internmc.facebook.com/intern/diff/D22938664)

[ghstack-poisoned]

ilia-cher commented:
(rebase)
